usr/sbin/p11sak/p11sak.c

```diff
@@ -1141,6 +1191,200 @@ static CK_ATTRIBUTE_TYPE col2type(int col)
     }
 }
 
+/**
+ *  Print in p11sak_defined_attrs.conf defined attributes in short format
+ */
+static CK_RV short_print_vendor() 
+{
+    CK_RV rc = CKR_OK;
+    int f;
+    FILE *fp;
+    struct ConfigBaseNode *cfg, *c, *name, *hex_string, *type;
+    struct ConfigStructNode *structnode;
+    char file_loc[1024] = P11SAK_DEFINED_ATTRS_LOCATION;
+
+    // get file location
+    cfg = NULL;
+    if (strcmp(P11SAK_DEFINED_ATTRS_LOCATION, "") == 1) {
```

Comment:
I do not understand the logic here.  What is `P11SAK_DEFINED_ATTRS_LOCATION` good for?  I see it hard-coded to the empty string in `p11sak.h`, so the `strcmp` will always be 0 and we have some dead branches here.  What am I missing?

Also, never compare the result of `strcmp` against 1.  The function is defined in relation to 0, i.e., result <0, ==0, >0.  But is is also allowed to return, e.g., 2.

Reply:
Done